experimental/dds/tree2/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -9,41 +9,83 @@ import {
 	done,
 	IRandom,
 	createWeightedGenerator,
-	Weights,
+	BaseFuzzTestState,
 } from "@fluid-internal/stochastic-test-utils";
 import { DDSFuzzTestState } from "@fluid-internal/test-dds-utils";
 import { ISharedTreeView, SharedTreeFactory } from "../../../shared-tree";
 import { brand, fail } from "../../../util";
+import { FieldKey, FieldUpPath, JsonableTree, UpPath } from "../../../core";
+import { DownPath, TreeNode, toDownPath } from "../../../feature-libraries";
 import {
-	CursorLocationType,
-	FieldKey,
-	FieldUpPath,
-	moveToDetachedField,
-	UpPath,
-} from "../../../core";
-import {
-	FieldEdit,
 	FieldEditTypes,
-	FuzzDelete,
 	FuzzInsert,
+	FuzzSet,
 	FuzzTransactionType,
 	FuzzUndoRedoType,
 	Operation,
-	OptionalFieldEdit,
 	RedoOp,
-	SequenceFieldEdit,
+	Synchronize,
 	TransactionAbortOp,
 	TransactionBoundary,
 	TransactionCommitOp,
 	TransactionStartOp,
 	TreeEdit,
 	UndoOp,
 	UndoRedo,
-	ValueFieldEdit,
 } from "./operationTypes";
+import { FuzzNode, fuzzNode, fuzzViewFromTree, getEditableTree } from "./fuzzUtils";
 
 export type FuzzTestState = DDSFuzzTestState<SharedTreeFactory>;
 
+/**
+ * When performing an operation, a random field must be selected. Rather than enumerate all fields of the tree, this is
+ * performed recursively starting at the root field.
+ *
+ * When a field needs to be selected, the fuzz test generator walks down the tree from the root field, randomly selecting
+ * one of the below options. If the selected option is not valid (e.g. the field is empty and the generator is trying to
+ * select a node to delete), field selection will automatically re-sample excluding this option.
+ *
+ * Each weight is used throughout the field selection process.
+ *
+ * @remarks
+ * Allowing more than just numbers here could be interesting. E.g. `(depth: number) => number` to allow biasing away from
+ * changing the root field for more interesting merge outcomes. The `filter` parameter can already accomplish this, but is
+ * a bit less efficient in doing so (it'd need to walk to the root repeatedly to compute depth)
+ */
+export interface FieldSelectionWeights {
+	/**
+	 * Select the current Fuzz node's "optionalF" field
```

Comment:
If F stands for field, maybe we should just say field? I find it slightly confusing as just F.

Reply:
![image](https://github.com/microsoft/FluidFramework/assets/5222607/087ffaa6-7366-4ea3-8d62-c0b0b8b8161b)